install.go

```
@@ -542,3 +547,65 @@ func completeFileName(dir, name string) (string, error) {
+// asks the user whether yay should try to import them.
+func checkGpgKeys(srcinfos map[string]*gopkg.PKGBUILD) error {
+	// keys maps a given PGP with a boolean to indicate whether we would need to
+	// import this particular key.
+	keys := map[string]bool{}
+	// args will be usde to check all the keys at once.
+	args := []string{"--list-keys", "--with-sig-check"}
+
+	for _, pkgbuild := range srcinfos {
+		for _, key := range pkgbuild.Validpgpkeys {
+			// Using ToUpper to make sure case won't be an issue.
+			keys[strings.ToUpper(key)] = false
+			// For this check, it is OK if there are repeated keys.
+			args = append(args, key)
+		}
+	}
+
+	cmd := exec.Command(config.GpgBin, args...)
+	err := cmd.Run()
+
+	if err != nil {
+		// There was an issue with at least one key, so let's find out which
+		// ones gave us problems and we can offer to import them.
+		problematic := []string{}
+		for key := range keys {
+			args = []string{"--list-key", key}
+			cmd = exec.Command(config.GpgBin, args...)
+			err = cmd.Run()
+			if err != nil {
+				problematic = append(problematic, key)
+				keys[key] = true
+			}
+		}
+		keyStr := "key"
+		if len(problematic) > 1 {
+			keyStr = "keys"
+		}
+		question := fmt.Sprintf("Problem detected with the following %s: %q. Import?", keyStr, strings.Join(problematic[:], ","))
+		if continueTask(question, "nN") && !importKeys(problematic) {
+			return fmt.Errorf("Problem when trying to imports keys")
+		}
+	}
+
+	return nil
+}
+
+// importKeys tries to import the list of keys specified in its argument.
+func importKeys(keys []string) bool {
+	args := []string{"--recv-keys", strings.Join(keys[:], " ")}
+	cmd := exec.Command(config.GpgBin, args...)
+	cmd.Stdin, cmd.Stdout, cmd.Stderr = os.Stdin, os.Stdout, os.Stderr
+
+	fmt.Printf("%s Importing key(s) with gpg...\n", arrow)
+	err := cmd.Run()
+
+	if err != nil {
+		fmt.Printf("%s Problem importing key(s)\n", arrow)
+	}
+	return err == nil
```

Why not just return the error?